### PR TITLE
enable extractvalue functions when the `--bit-precise` flag is used

### PR DESCRIPTION
**Pull request by shaobo-he**
This PR addresses issue #441.

### CODE REVIEW

File: include/smack/SmackRep.h [hunk 1]
```diff
@@ -174,6 +173,7 @@ class SmackRep {
 
   void addAuxiliaryDeclaration(Decl* D);
   std::list<Decl*> auxiliaryDeclarations();
+  std::string opName(const std::string& operation, std::list<const llvm::Type*> types);
```
[the remark this answers is not in the document]
Comment (shaobo-he) [reply include/smack/SmackRep.h]: This function needs to be public.

File: lib/smack/SmackInstGenerator.cpp [hunk 2]
```diff
@@ -379,14 +388,18 @@ void SmackInstGenerator::visitShuffleVectorInst(ShuffleVectorInst &I) {
...
 void SmackInstGenerator::visitExtractValueInst(llvm::ExtractValueInst& evi) {
   processInstruction(evi);
-  if (!SmackOptions::BitPrecise) {
-    const Expr* e = rep->expr(evi.getAggregateOperand());
-    for (unsigned i = 0; i < evi.getNumIndices(); i++)
-      e = Expr::fn(Naming::EXTRACT_VALUE, e, Expr::lit((unsigned long long) evi.getIndices()[i]));
-    emit(Stmt::assign(rep->expr(&evi),e));
-  } else {
-    WARN("Ignoring extract instruction under bit vector mode.");
+  const Value* g = evi.getAggregateOperand();
```
[the remark this answers is not in the document]
Comment (shaobo-he) [reply lib/smack/SmackInstGenerator.cpp]: It was named like this before this PR. I can change it to a more meaningful name.

File: lib/smack/SmackInstGenerator.cpp [hunk 3]
```diff
@@ -379,14 +388,18 @@ void SmackInstGenerator::visitShuffleVectorInst(ShuffleVectorInst &I) {
...
+  const Expr* e = rep->expr(g);
+  const unsigned num_indices = evi.getNumIndices();
+  const Type* t = g->getType();
+  auto indices = evi.getIndices();
+  for (unsigned i = 0; i < num_indices; i++) {
```
[the remark this answers is not in the document]
Comment (shaobo-he) [reply lib/smack/SmackInstGenerator.cpp]: No, it was like this before the PR so I didn't change it. I can change it to using iterators.

File: lib/smack/SmackInstGenerator.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (zvonimir) [reply lib/smack/SmackInstGenerator.cpp]: Please do.

File: lib/smack/SmackInstGenerator.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (zvonimir) [reply lib/smack/SmackInstGenerator.cpp]: Please do.

**Comment by zvonimir** [comment]
@shaobo-he : please squash and then I'll merge. Thx!

**Comment by shaobo-he** [comment]
> @shaobo-he : please squash and then I'll merge. Thx!

Done. Please merge it after it passes the CI.

**Comment by zvonimir** [comment]
Your commit message should have `Fixes #441` in it. Can you add that and force push again?

**Comment by shaobo-he** [comment]
> Your commit message should have `Fixes #441` in it. Can you add that and force push again?

Yes, done.

**Pull request merged**